### ISSUE TITLE
是否可以实现类似“adb root”这种命令

### ISSUE THREAD

**Issue by firmianay** [openatx/adbutils#48]
想要pull /system，但是需要root权限

**Comment by nblog** [comment]
[参考 adb root](https://github.com/Swind/pure-python-adb/blob/master/ppadb/command/transport/__init__.py#L179)

**Comment by nblog** [comment]
# wait seconds before ready
    def root(self):
        try:
            with self.adb_device._client._connect() as c:
                c.send_command("host:transport:" + self.adb_device.serial)
                c.check_okay()
                c.send_command("root:")
                c.check_okay()
                return c.read_until_close().startswith(("restarting adbd as root", "adbd is already running as root"))
        except:
            return False

**Comment by codeskyblue** [comment]
supported in 0.15.5

**Comment by yinkh** [comment]
use device.root() get this error on window10 paltform in adbutils==1.0.3

```
  File "D:\python_workspace\TestKit\adb\views.py", line 54, in save_dropbox
    a = device.root()
  File "D:\python_workspace\TestKit\venv\lib\site-packages\adbutils\_device.py", line 324, in root
    return c.read_string_block()
  File "D:\python_workspace\TestKit\venv\lib\site-packages\adbutils\_adb.py", line 118, in read_string_block
    size = int(length, 16)
ValueError: invalid literal for int() with base 16: 'rest'
```

**Comment by yinkh** [comment]
change return c.read_string_block() to return c.read_until_close() inside root will fix this bug,need pr?

**Comment by codeskyblue** [comment]
fix in 1.0.5

**Comment by yinkh** [comment]
confirm fixed! 
![image](https://user-images.githubusercontent.com/8170059/181003340-f6d71735-c5be-48c5-a40b-0bcaa6ed4bae.png)
